modules/apps/fragment/fragment-impl/src/main/java/com/liferay/fragment/internal/processor/PortletRegistryImpl.java

```diff
@@ -189,52 +192,85 @@ public void writePortletPaths(
 		}
 	}
 
-	@Reference(
-		cardinality = ReferenceCardinality.MULTIPLE,
-		policy = ReferencePolicy.DYNAMIC,
-		target = "(com.liferay.fragment.entry.processor.portlet.alias=*)"
-	)
-	protected void setPortlet(
-		javax.portlet.Portlet jxPortlet, Map<String, Object> properties) {
-
-		String alias = MapUtil.getString(
-			properties, "com.liferay.fragment.entry.processor.portlet.alias");
-		String portletName = MapUtil.getString(
-			properties, "javax.portlet.name");
-
-		_portletNames.put(alias, portletName);
-
-		Bundle bundle = FrameworkUtil.getBundle(jxPortlet.getClass());
-
-		BundleContext bundleContext = bundle.getBundleContext();
-
-		_serviceRegistrations.put(
-			alias,
-			bundleContext.registerService(
-				PortletAliasRegistration.class,
-				new PortletAliasRegistration() {
-				},
-				HashMapDictionaryBuilder.<String, Object>put(
-					"com.liferay.fragment.entry.processor.portlet.alias", alias
-				).build()));
-	}
+	@Activate
+	protected void activate(BundleContext bundleContext)
+		throws InvalidSyntaxException {
+
+		String filterString = StringBundler.concat(
+			"(&(", _ALIAS_KEY, "=*)(objectClass=",
+			javax.portlet.Portlet.class.getName(), "))");
+
+		_serviceTracker = new ServiceTracker<>(
+			bundleContext, bundleContext.createFilter(filterString),
+			new ServiceTrackerCustomizer
+				<javax.portlet.Portlet, javax.portlet.Portlet>() {
+
+				@Override
+				public javax.portlet.Portlet addingService(
+					ServiceReference<javax.portlet.Portlet> serviceReference) {
+
+					String alias = GetterUtil.getString(
+						serviceReference.getProperty(_ALIAS_KEY));
+
+					_portletNames.put(
+						alias,
+						GetterUtil.getString(
+							serviceReference.getProperty(
+								"javax.portlet.name")));
+
+					_serviceRegistrations.put(
+						alias,
+						bundleContext.registerService(
+							PortletAliasRegistration.class,
+							new PortletAliasRegistration() {
```

Comment:
@dantewang I would like to remove the registration of PortletAliasRegistration, could you please take a look at?